kpc/admin.py

```diff
@@ -1,7 +1,45 @@
+from django import forms
 from django.contrib import admin
+from django.contrib.admin.widgets import FilteredSelectMultiple
+
+from accounts.models import Profile
+
 from .models import Licensee
 
 
+class LicenseeAdminForm(forms.ModelForm):
+    contacts = forms.ModelMultipleChoiceField(
+            queryset=Profile.objects.all(),
+            required=False,
+            widget=FilteredSelectMultiple(
+                verbose_name='Contacts',
+                is_stacked=False
+            )
+    )
+
+    class Meta:
+        model = Licensee
+        fields = '__all__'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if self.instance and self.instance.pk:
+            self.fields['contacts'].initial = self.instance.contacts.all()
+
+    def save(self, commit=True):
+        licensee = super().save(commit=False)
+        if commit:
+            licensee.save()
+
+        if licensee.pk:
+            licensee.contacts.set(self.cleaned_data['contacts'])
+        self.save_m2m()
```

Comment:
Should this `set` and `save_m2m` be guarded by the `if commit`? I wonder if there'd be potentially unexpected DB changes if we called save with `commit=False`.

Reply:
TIL, Django's ModelAdmin _always_ calls `form.save` with `commit=False`.  Modified implementation incoming which I think handles this a little more gracefully.